BUGFIX / NEXT-7517 - sw_extends: Use correct template inheritance order

AndreasA opened the pull request:
<!--
Thank you for contributing to Shopware! Please fill out this description template to help us to process your pull request.

Please make sure to fulfil our contribution guideline (https://docs.shopware.com/en/shopware-platform-dev-en/contribution/contribution-guideline?category=shopware-platform-dev-en/contribution).

Do your changes need to be mentioned in the documentation?
Add notes on your change right now in the documentation files in /src/Docs/Resources and add them to the pull request as well. 
-->

### 1. Why is this change necessary?

To fix the issue https://issues.shopware.com/issues/NEXT-7517
Fixes: #614

Might also resolve: 
#707

### 2. What does this change do, exactly?
Sets the correct inheritence order if a source template is provided to the template finder.

### 3. Describe each step to reproduce the issue or behaviour.
See related issuess.

### 4. Please link to the relevant issues (if any).
https://issues.shopware.com/issues/NEXT-7517
https://github.com/shopware/platform/issues/614

Might also resolve:
https://github.com/shopware/platform/issues/707

soebbing commented:
Hi @samuelvogel, I just merged your PR as https://github.com/shopware/shopware/commit/8cc5963b92880551c2531d8f9a8e61ea45d54b19 - thank you so much for this fix! 💙

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/shopware/platform?pullRequest=1396) <br/>All committers have signed the CLA.

AndreasA committed:
BUGFIX: NEXT-7517

sw_extends: Use correct template inheritance order,
            if extending the template in the same plugin.

AndreasA commented:
@bneumann97 Any news regarding reviewing the PR? Also why did the bot create a new issue for this? There was already an issue?

bneumann97 commented:
@AndreasA We always create an issue for GitHub PR's for our internal workflow. The issues will be linked internally. There are several more PR's open longer than yours. As this is not a quick pick we have to take a lot of time for testing and reviewing. Please be patient 🙂

bneumann97 commented:
Hey @AndreasA, thanks for your contribution 💙

AndreasA commented:
@bneumann97 As the PR is merged I guess you did not find any issues and everything works fine?

Any idea about a possible release date for this fix - in an official version?

mitelg commented:
as the release train for Shopware 6.3.4.0 has already departed, it will be part of the next minor version 6.3.5.0

TaiKamilla commented:
Thank you!!

shyim commented:
Just as info:

We decided to make this change behind a Feature Flag. We don't want to deliver such a change in a minor update. This feature can be still enabled by a  env `FEATURE_NEXT_12553=1` and will be enabled by default on next major (6.4.0.0)

rickschippers commented:
Thanks, I ran into the same issue today. 
I have a theme in a bundle that  `sw_extends '@Storefront/storefront/base.html.twig'`.

Then I have another bundle that `sw_extends '@Storefront/storefront/base.html.twig'` in a new page template.  This uses the `base.html.twig` from shopware platform instead of from my theme bundle.

Using 6.3.5.0-dev with this feature enabled fixes the issue and loads the correct base template.

AndreasA commented:
Great to hear that the PR fixes the issue for others too 😄 
When can we expect the 6.3.5.0 release ?

mitelg commented:
the release is planned for the first Monday of February 😉
